protocol/msgpack/base_encoder.go

```diff
@@ -37,19 +37,34 @@ type encodeArrayLenFn func(value int)
 
 // baseEncoder is the base encoder that provides common encoding APIs.
 type baseEncoder struct {
-	bufEncoder       BufferedEncoder
-	encodeErr        error
-	encodePolicyFn   encodePolicyFn
-	encodeTimeFn     encodeTimeFn
-	encodeVarintFn   encodeVarintFn
-	encodeFloat64Fn  encodeFloat64Fn
-	encodeBytesFn    encodeBytesFn
-	encodeBytesLenFn encodeBytesLenFn
-	encodeArrayLenFn encodeArrayLenFn
+	bufEncoder               BufferedEncoder
+	policyCompressor         policy.Compressor
+	policyCompressionEnabled bool
+	encodeErr                error
+	encodePolicyFn           encodePolicyFn
+	encodeTimeFn             encodeTimeFn
+	encodeVarintFn           encodeVarintFn
+	encodeFloat64Fn          encodeFloat64Fn
+	encodeBytesFn            encodeBytesFn
+	encodeBytesLenFn         encodeBytesLenFn
+	encodeArrayLenFn         encodeArrayLenFn
 }
 
-func newBaseEncoder(encoder BufferedEncoder) encoderBase {
-	enc := &baseEncoder{bufEncoder: encoder}
+func newBaseEncoder(encoder BufferedEncoder, opts BaseEncoderOptions) encoderBase {
+	if opts == nil {
+		opts = NewBaseEncoderOptions()
+	}
+
+	compressor := opts.PolicyCompressor()
```

Comment:
Can we set the default compressor to `NoOpCompressor` so we don't need to do the nil check here?